python/semantic_kernel/orchestration/sk_function.py

```diff
@@ -152,12 +152,19 @@ async def _local_func(client, request_settings, context: "SKContext", **kwargs):
                 log.warning("Function call is not None, but functions is None")
             try:
                 if functions and hasattr(client, "complete_chat_with_functions_async"):
-                    (
-                        completion,
-                        function_call,
-                    ) = await client.complete_chat_with_functions_async(
+                    outputs = await client.complete_chat_with_functions_async(
                         messages, functions, request_settings
                     )
+                    if function_config.has_chat_with_data_prompt:
+                        completion, tool_message, function_call = outputs
+                        if tool_message:
+                            context.objects["tool_message"] = tool_message
```

Comment:
This looks good for now, for v1 we plan to get rid of the current way of working with the context, but for now this works.

Reply:
From the with-data side, the main thing we need is for the tool message to be in the chat message list (it contains the intent, which is used in future retrieval calls). I added it to the context in case it would need to be used there, but I can take it out if it's not needed.

Reply:
Understood, at some point I would like to introduce a class to hold responses so that we do not have to deal with this kind of ugly tuple, but that is for later!